x-pack/test/apm_api_integration/tests/cold_start/generate_data.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { apm, timerange } from '@elastic/apm-synthtrace';
+import type { ApmSynthtraceEsClient } from '@elastic/apm-synthtrace';
+
+export const dataConfig = {
+  serviceName: 'synth-go',
+  rate: 10,
+  transaction: {
+    name: 'GET /apple 🍎',
+    duration: 1000,
+  },
+};
+
+export async function generateData({
+  synthtraceEsClient,
+  start,
+  end,
+}: {
+  synthtraceEsClient: ApmSynthtraceEsClient;
+  start: number;
+  end: number;
+}) {
+  const { rate, transaction, serviceName } = dataConfig;
+  const instance = apm.service(serviceName, 'production', 'go').instance('instance-a');
+
+  const traceEvents = timerange(start, end)
+    .interval('1m')
+    .rate(rate)
```

Comment:
does it make sense to generate the same rate/duration for both? if we want to make sure data is properly filtered we should probably use different values for cold and warm starts.

Reply:
Good point, added different rates for the transactions